Week2-Section1

camodesigns opened the pull request:
Pull request for all the exercises of Month1-Week2-Section1

camodesigns committed:
Finished Exercise1

Contructor, Destructor and use of for whit the constructor

camodesigns committed:
Month1-Week2-Section1 Finished

This is de first commit of the First Section of the second week.
Exercise1
Constructor and Destructor creation and print
Exercise2
Military Clock
Use constructor
Set
Print
Exercise3
Shape Class for cicle and square
get data
void Area(), Perimeter;
input verifications
Print
Exercise4
Class String Printer
Constructor
Input Validation
void Print(),void PrintReverse(),void PrintSpaced(),void PrintVertical()

JDAlzate commented on Month1/Week2/Section1/Exercise1/main1.cpp:
```diff
@@ -0,0 +1,42 @@
+
+#include<string>
+
+class FirstClass
```
From now, we're following the standards. Your classes should use the `F` prefix. I know the instructions don't reflect this, but as stated on the standard page in Tettra, you still need to adjust it to follow the standard. 

The exercises will be adjusted to use the standard over time.

**Please fix this for all the exercises**

camodesigns replied:
Done

JDAlzate commented on Month1/Week2/Section1/Exercise1/main1.cpp:
```diff
@@ -0,0 +1,42 @@
+class FirstClass
+{
+public:
+	std::string Texto;
```
Remember all variables and comments should be in English. 

Please fix this for all exercises

camodesigns replied:
I change the word

JDAlzate commented on Month1/Week2/Section1/Exercise1/main1.cpp:
```diff
@@ -0,0 +1,42 @@
+	{
+
+	}
+	FirstClass(std::string texto)
```
Remember all variables and functions should use PascalCase. `texto` should be `Texto` (Though remember to also fix the name to make it English).

For conflicts between member names and constructor parameter names, it's acceptable to use the prefix `In` if you want. So you could call this `InText`, for example.

JDAlzate replied:
However, for this exercise, you should only have a default constructor and a default destructor, each with their appropriate text printed to the string. The `Texto` member and the `Print` member should not be included in the class, at it defeats the purpose of the exercise.

camodesigns replied:
Done

JDAlzate commented on Month1/Week2/Section1/Exercise1/main1.cpp:
```diff
@@ -0,0 +1,42 @@
+	}
+	void Print()
+	{
+		std::cout << Texto << std::endl;
```
When you call print, you're not actually in the constructor, so you should not be printing a message that says you are.

camodesigns replied:
Done

JDAlzate commented on Month1/Week2/Section1/Exercise1/main1.cpp:
```diff
@@ -0,0 +1,42 @@
+
+int main()
+{
+	FirstClass Construct("adsjksjkald");
```
You're supposed to do two things here:
1. Instantiate 5 classes individually (in 5 separate lines).
2. Instantiate one class inside the body of a loop that runs for 5 iterations.

Neither of those are happening here. You're just instantiating one object and calling a function on it 5 times.

camodesigns replied:
Done

JDAlzate commented on Month1/Week2/Section1/Exercise2/Main2.cpp:
```diff
@@ -0,0 +1,73 @@
+class MilitaryClock
+{
+private:
+	short hours ;
```
Variables should use PascalCase. **Please fix this for all the exercises**

JDAlzate replied:
There should be no space before the `;`

camodesigns replied:
Done

JDAlzate commented on Month1/Week2/Section1/Exercise2/Main2.cpp:
```diff
@@ -0,0 +1,73 @@
+		 hours=0;
+		 minutes=0;
+	}
+	MilitaryClock(short hour, short minute)
```
You should be following const-correctness. That means that your parameters should be `const`, since you're not changing their values.

camodesigns replied:
Done

JDAlzate commented on Month1/Week2/Section1/Exercise2/Main2.cpp:
```diff
@@ -0,0 +1,73 @@
+		 hours = hour;
+		 minutes = minute;
+	}
+	void SetTime(short sethours, short setminutes)
```
What's the point of requiring parameters if you're going to change their values to the same literals every time? This function is not doing what it's supposed to do.

camodesigns replied:
Done

JDAlzate commented on Month1/Week2/Section1/Exercise2/Main2.cpp:
```diff
@@ -0,0 +1,73 @@
+	}
+	void DisplayTime()
+	{
+		if (hours < 10 && minutes<10)
```
You're repeating conditions here to do the same thing. Maybe there is a way to use just 2 if's here 👀 

What if you try to format hours and minutes individually?

camodesigns replied:
I change the if in all the cases, I have only two now

JDAlzate commented on Month1/Week2/Section1/Exercise2/Main2.cpp:
```diff
@@ -0,0 +1,73 @@
+	clock.DisplayTime();
+	clock=MilitaryClock(7, 12);
+	clock.DisplayTime();
+	clock.SetTime(12, 15);
```
What if I tried to set the time to "9, 12"? This wouldn't be possible with the way your function is written.

camodesigns replied:
I change the class and this is posible now

JDAlzate commented on Month1/Week2/Section1/Exercise3/Main3.cpp:
```diff
@@ -0,0 +1,167 @@
+class Square
+{
+private:
+	float Side;
```
This name is fine, but it could be more descriptive. What _is_ Side? Is it the amount of sides the square has (for some reason)? Is it an index of which side you are currently evaluating? Or is it the length of the side? In that case, maybe `SideLength` is clearer?

camodesigns replied:
Done

JDAlzate commented on Month1/Week2/Section1/Exercise3/Main3.cpp:
```diff
@@ -0,0 +1,167 @@
+{
+private:
+	float Side;
+	float AreaSquare;
```
This is a descriptive name, but redundant. `Area` would already imply that it's the Square's area since it's inside the Square class. There is no need to have all variables specify that they are part of the square.

camodesigns replied:
Done

JDAlzate commented on Month1/Week2/Section1/Exercise3/Main3.cpp:
```diff
@@ -0,0 +1,167 @@
+	Square()
+	{
+	}
+	Square(float Radius, float Diameter)
```
Why does a square need a radius and a diameter? Why does it not use it for anything?

JDAlzate commented on Month1/Week2/Section1/Exercise3/Main3.cpp:
```diff
@@ -0,0 +1,167 @@
+	float PerimeterSquare;
+public:
+	Square()
+	{
```
Your default constructor should default-initialize your variables. None of your variables are initialized here.

camodesigns replied:
Done

JDAlzate commented on Month1/Week2/Section1/Exercise3/Main3.cpp:
```diff
@@ -0,0 +1,167 @@
+		std::cout << "what is the side of Your Square:" << std::endl;
+		std::cin >> side;
+
+		while (!std::cin.good() || side < 0)
```
Remember floats use an f at the end

JDAlzate commented on Month1/Week2/Section1/Exercise3/Main3.cpp:
```diff
@@ -0,0 +1,167 @@
+			std::cin.ignore(1000, '\n');
+			std::cin >>side;
+		}
+		Side = side;
```
Variables should make logical sense in their operations, that's why they need descriptive names. What would you think is happening here if you only read this line?

camodesigns replied:
Done

JDAlzate commented on Month1/Week2/Section1/Exercise3/Main3.cpp:
```diff
@@ -0,0 +1,167 @@
+		}
+		Side = side;
+	}
+	float GetAreaSquare( )
```
Why do you have a space inside the function parentheses?

camodesigns replied:
Done

JDAlzate commented on Month1/Week2/Section1/Exercise3/Main3.cpp:
```diff
@@ -0,0 +1,167 @@
+	}
+	float GetAreaSquare( )
+	{
+		AreaSquare = pow(Side, 2);
```
You're assigning a value for this member each time you're calling the method. With that in mind... is it really necessary to have a member for area? Why not just have the function return the value directly? And, in that case, you could make your function const. What do you think?

This applies for both area and perimeter.

camodesigns replied:
Done

JDAlzate commented on Month1/Week2/Section1/Exercise3/Main3.cpp:
```diff
@@ -0,0 +1,167 @@
+		PerimeterSquare = Side*4;
+		return PerimeterSquare;
+	}
+	void PrinterSquare() 
```
Remember, functions represent _actions_ so they should be named accordingly and start with a verb. The word _Printer_ is a noun, so it's not describing an action.

camodesigns replied:
Done

JDAlzate commented on Month1/Week2/Section1/Exercise3/Main3.cpp:
```diff
@@ -0,0 +1,167 @@
+	float Radius;
+	float Diameter;
+public:
+	Circle()
```
Same comment here about the default constructor.

camodesigns replied:
Done

JDAlzate commented on Month1/Week2/Section1/Exercise3/Main3.cpp:
```diff
@@ -0,0 +1,167 @@
+	}
+	Circle(float Radius, float Diameter)
+	{
+		Radius = 0;
```
Why are you taking in parameters if you are just going to initialize your variables to 0 anyway?

camodesigns replied:
Done

JDAlzate commented on Month1/Week2/Section1/Exercise3/Main3.cpp:
```diff
@@ -0,0 +1,167 @@
+	{
+		Side = 0;
+	}
+	void DataSquare()
```
If you read this function name, would you know what it is doing?

camodesigns replied:
Done

JDAlzate commented on Month1/Week2/Section1/Exercise3/Main3.cpp:
```diff
@@ -0,0 +1,167 @@
+		Radius = 0;
+		Diameter = 0;
+	}
+	void DataCircle()
```
Same comment here about the function name.

camodesigns replied:
Done

JDAlzate commented on Month1/Week2/Section1/Exercise3/Main3.cpp:
```diff
@@ -0,0 +1,167 @@
+			std::cin.ignore(1000, '\n');
+			std::cin >> diameter;
+		}
+		Diameter=diameter;
```
Same comment here about the names. Also, remember the spaces between operators

camodesigns replied:
Done

JDAlzate commented on Month1/Week2/Section1/Exercise3/Main3.cpp:
```diff
@@ -0,0 +1,167 @@
+		Diameter=diameter;
+	
+	}
+	float GetAreaCicle( )
```
Same comment here about the space in the function parentheses. Same for the next function.

camodesigns replied:
Done

JDAlzate commented on Month1/Week2/Section1/Exercise3/Main3.cpp:
```diff
@@ -0,0 +1,167 @@
+	}
+	float GetAreaCicle( )
+	{
+		AreaCircle =M_PI*(pow(Radius,2));
```
Same comment about the member variables.

camodesigns replied:
Done

JDAlzate commented on Month1/Week2/Section1/Exercise3/Main3.cpp:
```diff
@@ -0,0 +1,167 @@
+		PerimeterCircle = M_PI * Diameter;
+		return PerimeterCircle;
+	}
+	void PrinterCircle()
```
Same comment here about the name.

camodesigns replied:
Done

JDAlzate commented on Month1/Week2/Section1/Exercise4/Main4.cpp:
```diff
@@ -0,0 +1,88 @@
+	}
+	Printer(std::string OutputString)
+	{
+		OutputString = "";
```
Avoid hiding your data members. You have the same name for the member variable and the parameter variable.

Also, why take in a parameter when you are not going to use it?

camodesigns replied:
Done

JDAlzate commented on Month1/Week2/Section1/Exercise4/Main4.cpp:
```diff
@@ -0,0 +1,88 @@
+			std::cin.ignore(1000, '\n');
+			std::cin >> Outputstring;
+		}
+		OutputString = Outputstring;
```
Capitalization alone is not an acceptable way to differentiate variables. It's hard to see that there is even a difference and there is no conceptual difference between the variables.

Also, it violates our naming standards, which require PascalCase.

camodesigns replied:
Done

JDAlzate commented on Month1/Week2/Section1/Exercise4/Main4.cpp:
```diff
@@ -0,0 +1,88 @@
+		}
+		std::cout << std::endl;
+	}
+	void PrintSpaced() 
```
This function is supposed to take in the number of spaces you want between characters.

camodesigns replied:
Done

JDAlzate reviewed (changes requested):
Review is up! :)

camodesigns committed:
Month1-Week2-Section1 Updates

Variables and const names changed
Change method Exercise1
Standars Applied

camodesigns committed:
Change a calls of methods

camodesigns committed:
Month1-Week2-Section1Updates Exercise 2 and 3

Spaces updated
Names Update

JDAlzate commented on Month1/Week2/Section1/Exercise1/main1.cpp:
```diff
@@ -8,35 +8,39 @@ Now create a loop of at least five iterations. For each iteration, instantiate y
-	FirstClass(std::string texto)
-	{
-		Texto = "I�m in the constructor!";
+		ConstructorOutput = "I am printing the constructor";
```
You're never printing this text. The idea is for the constructor to print the string. Do you really need a variable for this?

camodesigns replied:
Done

JDAlzate commented on Month1/Week2/Section1/Exercise1/main1.cpp:
```diff
@@ -8,35 +8,39 @@ Now create a loop of at least five iterations. For each iteration, instantiate y
 	}
-	void Print()
+};
+void FFirstTest()
```
Methods should not use the `F` prefix. This is just for class and struct declarations.

camodesigns replied:
Changed

JDAlzate commented on Month1/Week2/Section1/Exercise2/Main2.cpp:
```diff
@@ -18,56 +18,55 @@ Example output
+		 Minutes=0;
 	}
-	MilitaryClock(short hour, short minute)
+	FMilitaryClock(short const InitialHours, short const InitialMinutes)
```
Though const can be added before or after the type declaration, we prefer to use it before. That's my bad for not adding it to the conventions, I will do that now. But please change it to be `const short` rather than `short const` please :)
This goes for the whole review 😅

camodesigns replied:
Changed

JDAlzate commented on Month1/Week2/Section1/Exercise2/Main2.cpp:
```diff
@@ -18,56 +18,55 @@ Example output
-		hours = sethours;
-		minutes=setminutes;
+		Hours = SetHours;
+		Minutes= SetMinutes;
```
Could you add a space before that =?

camodesigns replied:
Added

JDAlzate commented on Month1/Week2/Section1/Exercise2/Main2.cpp:
```diff
@@ -18,56 +18,55 @@ Example output
 	void DisplayTime()
 	{
-		if (hours < 10 && minutes<10)
+		if ((Hours) < 10)
```
Any reason to add the parentheses to Hours? Doesn't seem necessary.

camodesigns replied:
parentheses eliminated

JDAlzate commented on Month1/Week2/Section1/Exercise2/Main2.cpp:
```diff
@@ -18,56 +18,55 @@ Example output
-		}
-		else if (hours<10 && minutes>=10) 
+			std::cout << "0" << Hours;
+		}else
```
Any new statement should be on its own line. There shouldn't be any code after the closing brace of a code block.

JDAlzate commented on Month1/Week2/Section1/Exercise2/Main2.cpp:
```diff
@@ -18,56 +18,55 @@ Example output
+			std::cout << Hours;
 		}
-		else if (hours>=10 && minutes<10) 
+		 if ( Minutes < 10) 
```
Please remove that extra space after the `(`

JDAlzate commented on Month1/Week2/Section1/Exercise2/Main2.cpp:
```diff
@@ -18,56 +18,55 @@ Example output
 	{
-		 hours=0;
-		 minutes=0;
+		 Hours=0;
```
Mind your spaces between operators, please.

JDAlzate commented on Month1/Week2/Section1/Exercise3/Main3.cpp:
```diff
@@ -27,110 +27,95 @@ Your shape
 			std::cin.clear();
 			std::cin.ignore(1000, '\n');
-			std::cin >>side;
+			std::cin >>EntrySide;
```
There's a missing space here, too.

JDAlzate commented on Month1/Week2/Section1/Exercise3/Main3.cpp:
```diff
@@ -27,110 +27,95 @@ Your shape
+		SideLenght = EntrySide;
 	}
-	float GetAreaSquare( )
+	float GetAreaSquare()
```
This method could be const. Same as the next one.

JDAlzate replied:
Your methods should do what they _imply_ they do and should not have side effects. These methods are `getter` methods, meaning that when you read this method you expect them to return a value, not print anything to the screen. With that in mind, these getter methods should not be the ones responsible for printing anything to the screen, both for this shape and the circle shape.

JDAlzate commented on Month1/Week2/Section1/Exercise3/Main3.cpp:
```diff
@@ -27,110 +27,95 @@ Your shape
 	float GetPerimeterSquare()
 	{
-		PerimeterSquare = Side*4;
+		float const PerimeterSquare = SideLenght *4;
```
Mind your spaces and remember floats have an f at the end.

JDAlzate commented on Month1/Week2/Section1/Exercise3/Main3.cpp:
```diff
@@ -27,110 +27,95 @@ Your shape
+	float SideLenght;
 public:
-	Square()
+	FSquare()
```
Careful with the exercise requirements. You're supposed to have a default constructor and a constructor that takes in parameters.

JDAlzate commented on Month1/Week2/Section1/Exercise3/Main3.cpp:
```diff
@@ -27,110 +27,95 @@ Your shape
+	float DiameterCircle;
 public:
-	Circle()
+	FCircle()
```
Same thing here about the constructor

JDAlzate commented on Month1/Week2/Section1/Exercise3/Main3.cpp:
```diff
@@ -27,110 +27,95 @@ Your shape
-		while (!std::cin.good() || diameter < 0)
+		std::cout << "what is the diameter of Your Circle:" << std::endl;
+		std::cin >> EntryDiameter;
+		while (!std::cin.good() || EntryDiameter < 0)
```
Missing .f here

JDAlzate commented on Month1/Week2/Section1/Exercise3/Main3.cpp:
```diff
@@ -27,110 +27,95 @@ Your shape
 		}
-		Diameter=diameter;
-	
+		DiameterCircle=EntryDiameter;
```
Missing spaces here.

JDAlzate commented on Month1/Week2/Section1/Exercise3/Main3.cpp:
```diff
@@ -27,110 +27,95 @@ Your shape
-	
+		DiameterCircle=EntryDiameter;
 	}
 	float GetAreaCicle( )
```
This method and the next could be const. Also, there's still spaces there. You commented `Done` but you didn't address the comment 👀

JDAlzate commented on Month1/Week2/Section1/Exercise3/Main3.cpp:
```diff
@@ -27,110 +27,95 @@ Your shape
 	float GetAreaCicle( )
 	{
-		AreaCircle =M_PI*(pow(Radius,2));
+		float AreaCircle = M_PI * (pow(RadiusCircle,2));
```
AreaCircle and PerimeterCircle could be const.

JDAlzate commented on Month1/Week2/Section1/Exercise3/Main3.cpp:
```diff
@@ -150,18 +135,16 @@ int main()
-		Circle circle;
-		circle.DataCircle();
+		FCircle circle;
+		circle.InPutDataCircle();
```
Be careful to maintain consistency. Your circle method is called `InPutDataCircle` and the one in the square is called `InputDataSquare`, without an uppercase `P`. The correct way of spelling it is without the uppercase `P`, Input is just one word.

JDAlzate commented on Month1/Week2/Section1/Exercise3/Main3.cpp:
```diff
@@ -150,18 +135,16 @@ int main()
 	{
-		Circle circle;
-		circle.DataCircle();
+		FCircle circle;
```
Remember we use PascalCase. `circle` is not a correct name.

camodesigns replied:
Applied

JDAlzate commented on Month1/Week2/Section1/Exercise3/Main3.cpp:
```diff
@@ -150,18 +135,16 @@ int main()
 	{
-		Square square;
-		square.DataSquare();
+		FSquare square;
```
Same thing here. `square` does not follow convention.

camodesigns replied:
changed

JDAlzate commented on Month1/Week2/Section1/Exercise4/Main4.cpp:
```diff
@@ -23,31 +23,28 @@ o
-	{
-		OutputString = "";
-	}
+	
```
Why are you removing the other constructors? They are required by the exercise specifications.

camodesigns replied:
Added

JDAlzate commented on Month1/Week2/Section1/Exercise4/Main4.cpp:
```diff
@@ -23,31 +23,28 @@ o
+		std::string EntryString;
 		std::cout << "Welcome! Please enter a string:" << std::endl;
-		std::getline( std::cin,Outputstring);
+		std::getline( std::cin,EntryString);
```
Spaces are not following convention here.

camodesigns replied:
Changed

JDAlzate commented on Month1/Week2/Section1/Exercise4/Main4.cpp:
```diff
@@ -79,10 +81,10 @@ class Printer
 int main() 
 {
-	Printer string;
+	FPrinter string;
```
Why is your printer index named `string`? `string` implies that your object is an `std::string` not an `FPrinter`. Also, please follow PascalCase 👀

camodesigns replied:
Renamed

JDAlzate reviewed (changes requested):
Comments added! :)

camodesigns committed:
Month1-Week2-Section Updates

Constructors
Spaces
Methods
Change const of place

JDAlzate commented on Month1/Week2/Section1/Exercise3/Main3.cpp:
```diff
@@ -31,10 +31,17 @@ class FSquare
+	FSquare() 
+	{
+	}
+	FSquare(float SideLenght, float PerimeterSquare, float AreaSquare)
```
Your parameter variables should not be named exactly like your member variables, since now your parameters are overshadowing your members. If you do anything to SideLength here, you will be changing the parameter value, not the member.

Apart from that, I have 3 comments:
1. Why are you asking for parameters if you are going to assign to those parameters the value 0.f and do nothing else. The parameters of constructors should be assigned to your member variables.
2. Why are you asking for a Permiter and an Area? These are not values that the user should input, those are the ones that you calculate inside the class.
3. Your parameters could all be `const float`.

Please take a look at the Constructor video again https://www.youtube.com/watch?v=FXhALMsHwEY&ab_channel=TheCherno

If you have any doubt about what the purpose of a parameter is or what a constructor with parameters should be doing, please let me know! I'll gladly answer them :)

camodesigns replied:
Yes changed, thanks

JDAlzate commented on Month1/Week2/Section1/Exercise3/Main3.cpp:
```diff
@@ -49,21 +56,19 @@ class FSquare
 		SideLenght = EntrySide;
 	}
-	float GetAreaSquare()
+	const float GetAreaSquare()
```
This syntax says that the GetAreaSquare function is returning a `const float` which is not desirable, you shouldn't return const values from functions. Instead, what you want to do is make the method `const` and for that, you want to type const at the _end_ of the function declaration:
```
float GetAreaSquare() const
{

}
```

camodesigns replied:
changed and understanded

JDAlzate commented on Month1/Week2/Section1/Exercise3/Main3.cpp:
```diff
@@ -73,18 +78,21 @@ class FCircle
 	float DiameterCircle;
```
Remember, it's redundant to call your members stuff like `DiameterCircle` because the fact that it is part of the Circle class is already known. So the name `Diameter` would be enough.

camodesigns replied:
changed

JDAlzate commented on Month1/Week2/Section1/Exercise3/Main3.cpp:
```diff
@@ -73,18 +78,21 @@ class FCircle
 	FCircle()
+	{
+	}
+	FCircle(float RadiusCircle,float DiameterCircle)
```
Same comment here about the constructor of the square

camodesigns replied:
updated

JDAlzate commented on Month1/Week2/Section1/Exercise3/Main3.cpp:
```diff
@@ -73,18 +78,21 @@ class FCircle
 	{
 		float EntryRadius;
 		float EntryDiameter;
 		std::cout << "what is the area of Your Circle:" << std::endl;
```
I just noticed, you're asking for the area, not the radius.

camodesigns replied:
haha, sorry, but updated

JDAlzate commented on Month1/Week2/Section1/Exercise3/Main3.cpp:
```diff
@@ -73,18 +78,21 @@ class FCircle
-		std::cin >>EntryRadius;
-		while (!std::cin.good() || EntryRadius < 0)
+		std::cin >> EntryRadius;
+		while (!std::cin.good() || EntryRadius < 0.0f)
```
Why are you asking for a radius if you never use it?

camodesigns replied:
![image](https://user-images.githubusercontent.com/36427649/180801287-8ff9e022-b99e-4d1e-93e0-8c1fec48dc79.png)
 I use the radius here for find the area

JDAlzate commented on Month1/Week2/Section1/Exercise3/Main3.cpp:
```diff
@@ -95,26 +103,22 @@ class FCircle
+		DiameterCircle = EntryDiameter;
 	}
-	float GetAreaCicle( )
+	const float GetAreaCicle( )
```
Same comment about const.

camodesigns replied:
Updated

JDAlzate commented on Month1/Week2/Section1/Exercise3/Main3.cpp:
```diff
@@ -135,16 +139,21 @@ int main()
-		circle.GetPerimeterCircle();
+		FCircle Circle;
+		Circle.InputDataCircle();
+		Circle.GetAreaCicle();
```
Why call GetAreaCircle and GetPerimeterCircle here if you're not doing anything with those values?

camodesigns replied:
Unused GetArea and GetPerimeter Eliminated

JDAlzate commented on Month1/Week2/Section1/Exercise4/Main4.cpp:
```diff
@@ -31,12 +31,16 @@ class FPrinter
 	FPrinter()
 	{
 	}
+	FPrinter(std::string EntryString)
```
Same comment as the other constructors

camodesigns replied:
Constructor Modified

JDAlzate reviewed (changes requested):
Review is up!

camodesigns committed:
Moth1-Week2-Section1-Exercise4Update

Entry file update

camodesigns committed:
Month1-Week2-Section1Updates

Renames
Modify constructors  for a correct use
change uses of const

camodesigns committed:
Month1-Week2-Section1

Unusedmethods eliminated

camodesigns committed:
Month1-Week2-Section1 Update Verify Input

Add condition of verification

JDAlzate commented on Month1/Week2/Section1/Exercise4/Main4.cpp:
```diff
@@ -27,26 +27,28 @@ class FPrinter
-		std::getline( std:: cin, EntryString);
-		while(!std::cin.good())
+		std::getline(std::cin, EntryString);
+		while(!std::cin.good()||EntryString==" ")
```
Mind your spaces here. Remember that operators should have spaces between them except the `!`, `;`, `::`, and `.` operators.

camodesigns replied:
Ok  change the spaces

JDAlzate reviewed (changes requested):
Just one comment, considering that we discussed changes in-person :)

camodesigns committed:
Month1-Week2-Section1 Updares

Eliminate unnecesarries declarations
Eliminate innecesary value for Circle class

camodesigns committed:
Month1-Week2-Section1 Update

Spaces updates

JDAlzate reviewed (approved):
Nice!
